cli/commands/benchmark-web-vitals.mjs

```diff
@@ -242,117 +335,165 @@ async function benchmarkURL( browser, params ) {
-		if ( url.username && url.password ) {
+		if ( urlObj.username && urlObj.password ) {
 			await page.authenticate( {
-				username: url.username,
-				password: url.password,
+				username: urlObj.username,
+				password: urlObj.password,
 			} );
 		}
 
-		const response = await page.goto( url.toString(), {
+		const response = await page.goto( urlObj.toString(), {
 			waitUntil: 'networkidle0',
 		} );
-		await page.addScriptTag( { content: scriptTag, type: 'module' } );
+		if ( scriptTag ) {
+			await page.addScriptTag( { content: scriptTag, type: 'module' } );
+		}
 
 		if ( response.status() !== 200 ) {
 			continue;
 		}
 
 		completeRequests++;
 
-		await Promise.all(
-			Object.values( metricsDefinition ).map( async ( value ) => {
-				// Wait until global is populated.
-				await page.waitForFunction(
-					`window.${ value.global } !== undefined`
-				);
-
-				/*
-				 * Do a random click, since only that triggers certain metrics
-				 * like LCP, as only a user interaction stops reporting new LCP
-				 * entries. See https://web.dev/lcp/.
-				 *
-				 * Click off screen to prevent clicking a link by accident and navigating away.
-				 */
-				await page.click( 'body', { offset: { x: -500, y: -500 } } );
-				// Get the metric value from the global.
-				/** @type {number} */
-				const metric = await page.evaluate(
-					( global ) => /** @type {number} */ window[ global ],
-					value.global
-				);
-				value.results.push( metric );
-			} )
-		).catch( () => {
-			/* Ignore errors. */
-		} );
-	}
+		if ( groupedMetrics.webVitals ) {
+			await Promise.all(
+				Object.values( groupedMetrics.webVitals ).map(
+					async ( value ) => {
+						// Wait until global is populated.
+						await page.waitForFunction(
+							`window.${ value.global } !== undefined`
+						);
+
+						/*
+						 * Do a random click, since only that triggers certain metrics
+						 * like LCP, as only a user interaction stops reporting new LCP
+						 * entries. See https://web.dev/lcp/.
+						 *
+						 * Click off screen to prevent clicking a link by accident and navigating away.
+						 */
+						await page.click( 'body', {
+							offset: { x: -500, y: -500 },
+						} );
+						// Get the metric value from the global.
+						/** @type {number} */
+						const metric = await page.evaluate(
+							( global ) =>
+								/** @type {number} */ window[ global ],
+							value.global
+						);
+						value.results.push( metric );
+					}
+				)
+			).catch( () => {
+				/* Ignore errors. */
+			} );
+		}
 
-	const metrics = {};
-	Object.entries( metricsDefinition ).forEach( ( [ key, value ] ) => {
-		if ( value.results.length ) {
-			metrics[ key ] = value.results;
+		if ( groupedMetrics.serverTiming ) {
+			const serverTimingMetrics = await page.evaluate( () =>
+				performance
+					.getEntries()[ 0 ]
+					.serverTiming.reduce( ( acc, value ) => {
+						acc[ value.name ] = value.duration;
+						return acc;
+					}, {} )
+			);
```

Comment:
This should be more robust to ensure that the performance entry is indeed of the expected type:

```suggestion
			const serverTimingMetrics = await page.evaluate( () => {
				const entry = performance.getEntries().find(
					( entry ) => entry instanceof PerformanceNavigationTiming
				);
				if ( entry instanceof PerformanceNavigationTiming ) {
					return entry.serverTiming.reduce(
						( acc, value ) => {
							acc[ value.name ] = value.duration;
							return acc;
						}, {}
					);
				} else {
					return {};
				}
			} );
```